common-test/src/main/java/io/strimzi/test/k8s/BaseKubeClient.java

```diff
@@ -234,13 +239,12 @@ private K waitFor(String resource, String name, Predicate<JsonNode> ready) {
     }
 
     @Override
-    public K waitForDeployment(String name) {
-        return waitFor("deployment", name, actualObj -> {
+    public K waitForResourceReady(String resourceType, String name) {
+        return waitFor(resourceType, name, actualObj -> {
```

Comment:
This check here with `status.replicas` and `status.readyReplicas` isn't suitable for all resources, so I think sticking with `waitForDeployment()` makes sense as it prevent people trying to use this to wait for a service to be ready, for example.

Reply:
The problem here, that we will have the same code for resource type `deploymentConfig` that we already have for resource type `deployment` and it was a reason that I updated arguments for the existing method but I don't know that it is NOT suitable for all resources, so I think that I should create new one